### ISSUE TITLE
Enhancement: no-restricted-imports use NOT in importNames

### ISSUE THREAD

**Issue by zqstack** [typescript-eslint/typescript-eslint#5467]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-restricted-imports

### Description

As an option I would like to see a possibility to restrict "all imports except given names".

I propose a possiblity to use an exxclamation mark "!" in front of import names to exclude them from restriction.

Examples

```
"paths": [
        {
          "name": "react-native",
          "importNames": ["!!StyleSheet"],
          "message": "Only StyleSheet is allowed to be imported from react-native."
        }
],
```

### Fail

```typescript
import { View } from 'react-native';
```

### Pass

```typescript
const replace = 'me';
import { StyleSheet} from 'react-native';
```

**Comment by bradzacher** [comment]
Thanks for the issue!

This is an extension rule. That means we take the base rule and extend it to support TS syntax or features.

We don't break from the base rule to add new features unless they are specifically related to TS.

In this instance - this request is not related to TS, thus we shall not implement it. If you'd like to see it, feel free to raise a request with the ESLint team.

**Comment by zqstack** [comment]
Thank you for your response! I also have a question.

As I am experimenting a bit, how can I add my forked package using yarn to my project to test changes in my project?

I tried adding `zqstack/typescript-eslint/eslint-plugin`, but it either says 404 not found or package version not found.
